app/questionnaire/placeholder_parser.py

```diff
@@ -52,9 +52,15 @@ def _lookup_answer(self, answer_id: str, list_item_id: str = None):
                     else answer.value
                     for list_item in answer.value
                 ]
-            return (
-                escape(answer.value) if isinstance(answer.value, str) else answer.value
-            )
+
+            if isinstance(answer.value, str):
+                return escape(answer.value)
+            elif isinstance(answer.value, dict):
+                answer_dict = {}
```

Comment:
Do we need to create a new dict here? Could we update the existing dict instead?

Reply:
Removed extra dicts in both methods, thanks!